### ISSUE TITLE
`weave launch` takes ages after removing all containers

### ISSUE THREAD

**Issue by rade** [weaveworks/weave#2286]
```
$ weave launch
$ docker rm -f $(docker ps -aq)
737d7e6b0146
74b6ffbdf2f9
e0616b22752d
8683748d26a6
8a9bcfe85377
$ time weave launch

real	1m1.315s
user	0m0.020s
sys	0m0.004s
```

The docker daemon logs says
```
May 11 09:05:54 xps docker[1009]: time="2016-05-11T09:05:54.962452385+01:00" level=error msg="Post http://%2Frun%2Fdocker%2Fplugins%2Fweave.sock/Plugin.Activate: dial unix /run/docker/plugins/weave.sock: connect: connection refused"
May 11 09:05:54 xps docker[1009]: time="2016-05-11T09:05:54.966588377+01:00" level=warning msg="Unable to connect to plugin: /run/docker/plugins/weave.sock:/Plugin.Activate, retrying in 1s"
(repeats)
```

**Comment by rade** [comment]
Tom says "If I remove the sockets in /run/docker/plugins it unblocks".

**Comment by bboreham** [comment]
Don't do that then. 

We provide `weave stop` and `weave reset` to shut down cleanly. 

Note that Docker may try to talk to the socket even if you don't intend to use it, eg to find out if it's a volume plugin. 

We may be able to improve that with a manifest.

**Comment by bboreham** [comment]
This is roughly the same problem described in #1934

**Comment by bboreham** [comment]
The JSON files described at https://docs.docker.com/engine/extend/plugin_api/ do not appear to qualify the type of plugin.

**Comment by bboreham** [comment]
Raised https://github.com/docker/docker/issues/22653

**Comment by awh** [comment]
Closed by #2292.
